scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -283,27 +309,31 @@ def get_threshold_metrics(y_true, y_pred):
     model: pipeline.named_steps['classify']
     for model, pipeline in final_pipelines.items()
 }
-coef_df_dict = {
-    model: get_model_coefficients(classifier, model, covariates.columns)
+
+coef_df = pd.concat([
+    get_model_coefficients(classifier, model, covariates.columns)
     for model, classifier in final_classifiers.items()
-}
+])
 
 
-# In[19]:
+# In[28]:
 
 model = 'full'
 
+model_coef_df = coef_df[coef_df['feature_set'] == model]
+
 print('{:.1%} zero coefficients; {:,} negative and {:,} positive coefficients'.format(
-    (coef_df_dict[model].weight == 0).mean(),
-    (coef_df_dict[model].weight < 0).sum(),
-    (coef_df_dict[model].weight > 0).sum()
+    (model_coef_df.weight == 0).mean(),
+    (model_coef_df.weight < 0).sum(),
+    (model_coef_df.weight > 0).sum()
 ))
```

Comment:
I'd do something more along the lines of:

```python
pandas.crosstab(coef_df.model, numpy.sign(coef_df.weight))
```

See https://pandas.pydata.org/pandas-docs/stable/generated/pandas.crosstab.html. There are several args you can play with... `normalize`, `margins`